aten/src/ATen/ThreadLocalState.cpp

```diff
@@ -21,6 +21,7 @@ ThreadLocalState::ThreadLocalState(bool keep_grad_mode)
   if (keep_grad_mode_) {
     grad_mode_enabled_ = GradMode::is_enabled();
   }
+  python_mode_state_ = at::impl::PythonModeTLS::get_state();
```

Comment:
FYI: this will conflict with this stack: https://github.com/pytorch/pytorch/pull/63117 that is landing soon